backend/app/services/knowledge/orchestrator.py

```
@@ -610,6 +613,16 @@ def list_documents(
         if not kb:
             raise ValueError("Knowledge base not found or access denied")
 
+        # Consumer role cannot view document list
+        has_access, role, _, _ = knowledge_share_service.get_user_kb_permission(
+            db, knowledge_base_id, user.id
+        )
+        if has_access and role == ResourceRole.CONSUMER.value:
+            raise ValueError(
+                "Consumer role cannot view document list. "
+                "You can use this knowledge base via chat only."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)